x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -804,4 +824,49 @@ protected LogicalPlan rule(Limit plan) {
             return p;
         }
     }
+
+    public static class ReplaceRegexMatch extends OptimizerRules.ReplaceRegexMatch {
+
+        protected Expression regexToEquals(RegexMatch<?> regexMatch, Literal literal) {
+            return new Equals(regexMatch.source(), regexMatch.field(), literal);
+        }
+    }
+
+    private static class ReplaceAttributesWithExact extends OptimizerRules.OptimizerRule<LogicalPlan> {
```

Comment:
Only the fields that occur inside certain operations need to be replaced - doing this across all fields is going to have side effects. Fields inside sort, stats need to be changed; potentially those used inside filters (though that depends whether full text search is used on them or not).
If the list is too broad, the rule could look into the fields that need to be preserved as are and then change everything else to be exact (though again that might backfire).

Reply:
👍 I'm changing this to limit the scope to `OrderBy`, `Aggregate` and `Filter` plans. I'm including Filter since we have no full text search for now, and it would be a shame not pushing down filters when possible. Likely, we'll have to refine this rule in the future.